astroid/interpreter/_import/util.py

```diff
@@ -2,15 +2,60 @@
 # For details: https://github.com/PyCQA/astroid/blob/main/LICENSE
 # Copyright (c) https://github.com/PyCQA/astroid/blob/main/CONTRIBUTORS.txt
 
-try:
-    import pkg_resources
-except ImportError:
-    pkg_resources = None  # type: ignore[assignment]
-
-
-def is_namespace(modname):
-    return (
-        pkg_resources is not None
-        and hasattr(pkg_resources, "_namespace_packages")
-        and modname in pkg_resources._namespace_packages
-    )
+from __future__ import annotations
+
+import pathlib
+from functools import lru_cache
+from importlib.util import _find_spec_from_path
+
+
+def _is_setuptools_namespace(location: pathlib.Path) -> bool:
+    try:
+        with open(location / "__init__.py", "rb") as stream:
+            data = stream.read(4096)
+    except OSError:
+        return False
+    else:
+        extend_path = b"pkgutil" in data and b"extend_path" in data
+        declare_namespace = (
+            b"pkg_resources" in data and b"declare_namespace(__name__)" in data
+        )
+        return extend_path or declare_namespace
+
+
+@lru_cache(maxsize=4096)
+def is_namespace(modname: str) -> bool:
+    found_spec = None
+
+    # find_spec() attempts to import parent packages when given dotted paths.
+    # That's unacceptable here, so we fallback to _find_spec_from_path(), which does
+    # not, but requires instead that each single parent ('astroid', 'nodes', etc.)
+    # be specced from left to right.
+    processed_components = []
+    last_parent = None
+    for component in modname.split("."):
+        processed_components.append(component)
+        working_modname = ".".join(processed_components)
+        try:
+            found_spec = _find_spec_from_path(working_modname, last_parent)
+        except ValueError:
+            # executed .pth files may not have __spec__
+            return True
+        last_parent = working_modname
+
+    if found_spec is None:
+        return False
+
+    if found_spec.origin == "namespace":
+        return True
```

Comment:
This is uncovered and based on https://docs.python.org/3/library/importlib.html#importlib.machinery.ModuleSpec.origin I think it should be `origin is None`.

Reply:
Right, this changed in 3.7 and I grabbed the wrong one. See https://github.com/PyCQA/astroid/blob/24b0bc346f485f10cff145b07507057dd72629fa/astroid/interpreter/_import/spec.py#L230-L233

I will open a separate cleanup for that quoted usage.

Reply:
Taking a look at the failed test...